src/cmd/turnkey/pkg/generate.go

```diff
@@ -40,9 +47,31 @@ var apiKeyCmd = &cobra.Command{
 			OutputError(eris.Wrap(err, "failed to read API key name"))
 		}
 
-		apiKey, err := apikey.New(Organization)
+		curveType, err := cmd.Flags().GetString("curve")
 		if err != nil {
-			OutputError(eris.Wrap(err, "failed to create API keypair"))
+			OutputError(eris.Wrap(err, "failed to read curve type"))
+		}
+
+		var apiKey *apikey.Key
+
+		switch curveType {
+		default:
+			OutputError(fmt.Errorf("invalid curve type: %s; supported types are p256, secp256k1, and ed25519", curveType))
+		case string(apikey.CurveP256):
+			apiKey, err = apikey.New(Organization, apikey.WithScheme(apikey.SchemeP256))
+			if err != nil {
```

Comment:
nit: could these copied err checks be moved to outside the switch statement so theres just 1 instead of 3 (plus more when there are more curve types)